src/controller/mazeCompression.java

```diff
@@ -0,0 +1,286 @@
+
+    private String mazeToHuffStr(maze2d maze) {
+
+        final StringBuilder sb = new StringBuilder();
+
+        sb.append(maze.mazeName);
+        sb.append("-");
+        sb.append(maze.getEntrance().x);
+        sb.append(",");
+        sb.append(maze.getEntrance().y);
+        sb.append("-");
+        sb.append(maze.getExit().x);
+        sb.append(",");
+        sb.append(maze.getExit().y);
+        sb.append("-");
+        sb.append(maze.mazeSize);
+        sb.append("-");
+        for (int i = 0; i < maze.mazeStructure[0].length; i++) {
+            for (int j = 0; j < maze.mazeStructure[1].length; j++) {
+                if (maze.mazeStructure[i][j])
+                    sb.append(1);
+                else
+                    sb.append(0);
+            }
+        }
+        return sb.toString();
+    }
+
+    public maze2d huffStrToMaze(String str) {
+
+        String[] splitStr = str.split("-");
+
+        int size = Integer.parseInt(splitStr[3]);
+
+        boolean [][] structure = new boolean[size][size];
+
+        int structureIdx = 0;
+        for (int i=0; i<size; i++) {
+            for (int j=0; j<size; j++) {
+                structure[i][j] = splitStr[4].charAt(structureIdx) == '1';
+                structureIdx++;
+            }
+        }
+
+
+        return new maze2d(splitStr[0], Integer.parseInt(splitStr[3]),
+                new Point(Integer.parseInt(splitStr[1].split(",")[0]),
+                        Integer.parseInt(splitStr[1].split(",")[1])),
+                new Point(Integer.parseInt(splitStr[2].split(",")[0]),
+                        Integer.parseInt(splitStr[2].split(",")[1])), structure);
+    }
+
+
+
+    public String encodeHuffmanAndSave(maze2d currMaze) throws IOException {
+        return encodeHuffmanAndSave(currMaze, null);
+    }
+
+
+    // encoding using Huffman and saving the
+    public String encodeHuffmanAndSave(maze2d currMaze, String mazeName) throws IOException {
+
+        String compressedMazeFilename;
+
+        if (mazeName == null)
+            compressedMazeFilename = currMaze.getClass().getSimpleName() + "@" + getUpdateMazeCnt() + ".bin";
+        else {
+            compressedMazeFilename = mazeName + ".bin";
+        }
+
+        // Converting compressor to string and encoding:
+        String huffInputStr = mazeToHuffStr(currMaze);
+        String actualEncoded = huffmanEncoder(huffInputStr);
+
+        // Creating binary file
+        DataOutputStream dos = new DataOutputStream(new FileOutputStream(compressedMazeFilename));
+
+        dos.writeBytes(actualEncoded);
+        dos.close();
+
+        return compressedMazeFilename;
+    }
+
+
+    public maze2d decodeHuffmanMazeFileToMaze(String encodedFilename) throws IOException {
+
+        String actualDecoded;
+        DataInputStream ois = new DataInputStream(new FileInputStream(encodedFilename));
+
+        byte[] binData = ois.readAllBytes();
+        String retrievedString = new String(binData);
+        ois.close();
```

Comment:
It is always a best practice to close a resource as soon as you finish using it
You don't have real exception handling here and if line 279 fails you'll have an un-closed file stuck..
You can move this one line up

Reply:
Fixed, THANKS!